src/clientversion.cpp

```diff
@@ -116,17 +116,46 @@ std::string FormatSubVersion(const std::string &name, int nClientVersion, const
     if (!subverOverride.empty())
         return subverOverride;
 
+    // sanitize comments per BIP014, format user agent and check total size
+    std::vector<std::string> uacomments = {};
+    for (std::string &cmt : mapMultiArgs["-uacomment"])
+    {
+        if (cmt != SanitizeString(cmt, SAFE_CHARS_UA_COMMENT))
+            break;
+        uacomments.push_back(SanitizeString(cmt, SAFE_CHARS_UA_COMMENT));
+    }
+    // If this is a 32bit build then append an identifier since we'd like to know
+    // how many still run this configuration.
+    {
+        int temp = 0;
+        int *ptemp = &temp;
+        if (sizeof(ptemp) == 4)
+            uacomments.push_back("32bit");
+    }
+
+    std::vector<std::string> vTotComments = comments;
+    vTotComments.insert(std::end(vTotComments), std::begin(uacomments), std::end(uacomments));
+
     std::ostringstream ss;
     ss << "/";
     ss << name << ":" << FormatVersion(nClientVersion);
-    if (!comments.empty())
+    if (!vTotComments.empty())
     {
-        std::vector<std::string>::const_iterator it(comments.begin());
+        std::vector<std::string>::const_iterator it(vTotComments.begin());
         ss << "(" << *it;
-        for (++it; it != comments.end(); ++it)
+        for (++it; it != vTotComments.end(); ++it)
             ss << "; " << *it;
         ss << ")";
     }
     ss << "/";
-    return ss.str();
+
+    std::string subver = ss.str();
+    if (subver.size() > MAX_SUBVERSION_LENGTH)
+    {
+        LOGA("Total length of network version string (%i) exceeds maximum length (%i). Reduce "
+             "the number or size of uacomments.",
+            subver.size(), MAX_SUBVERSION_LENGTH);
```

Comment:
fair point, I guess we should add a category for that logging statement. Not sure which one, thou. I'd say that NET is the best match even if it's not perfect. 

We could even imagine to remove the log message entirely since we are going to adding it back on `init.cpp` if the computed subver is equal to MAX_SUBVERSION_LENGTH as you suggested further down in the review.

I think will do the latter.